curvlinops/inverse.py

```diff
@@ -318,4 +336,44 @@ def _matmat(self, M: ndarray) -> ndarray:
                         ggT_inv, M_torch[pos], "i j, m j ... -> m i ..."
                     )
 
+        if return_tensor:
+            M_torch = cat([rearrange(M, "k ... -> (...) k") for M in M_torch], dim=0)
+
+        return M_torch
+
+    def torch_matvec(
+        self, v_torch: Tensor, return_tensor: bool = True
+    ) -> Union[Tensor, List[Tensor]]:
+        """Apply the inverse of KFAC to a vector in PyTorch.
+
+        This allows for matrix-vector products with the inverse KFAC approximation in
+        PyTorch without converting tensors to numpy arrays, which avoids unnecessary
+        device transfers when working with GPUs.
+
+        Args:
+            v_torch: Vector for multiplication. Has shape ``[D]``.
```

Comment:
I mainly introduced this option for `torch_matmat` so that I can use it within `_matmat`. But I think it is actually a cool feature to also support this here, since it will make gradient preconditioning more efficient (could also be done via `torch_matmat`, but I guess it's cleaner to directly support it with `torch_matvec`).

Reply:
Added support for list inputs for `torch_matvec`.